community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -389,7 +390,19 @@ public void force() throws IOException
         synchronized ( logFile )
         {
             writer.emptyBufferIntoChannelAndClearIt();
+        }
+        // Force the writer outside of the lock.
+        // This allows multiple threads forcing at the same time to piggy-back onto one another.
```

Comment:
Sorry for commenting on an already closed issue, but I have a long standing fondness for these code-paths.

I think the explanation here is wrong, because there is a forceLock.tryLock() two calls up in the call chain, only allowing a single thread in. So the performance improvement should be because you now allow new content to be written into the buffer even while the channel is being forced. Which also makes me wonder whether that buffer is rightly sized at a static 512K.

There is also a long comment further above that no longer makes any sense.